cobyla/cobyla.c

```diff
@@ -1505,6 +1505,12 @@ static nlopt_result trstlp(int *n, int *m, double *a,
 /* new value of ZDOTA(NACT) and branch if it is not acceptable. */
 
   i__1 = nact;
+  
+/* This pragma fixes a known problem compiling with VS2013 or VS2015 in Release */
+/* see https://connect.microsoft.com/VisualStudio/feedback/details/1028781/c1001-on-release-build */
+#if defined(_MSC_VER) && (_MSC_VER == 1800 || _MSC_VER == 1900)
```

Comment:
I would just use `#if defined(_MSC_VER)` here.  Who knows when this will be fixed, and really doesn't hurt to disable this optimization for this particular loops in all VC++ versions.